pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -832,29 +843,26 @@ func saveHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) (dirty boo
 	return
 }
 
-func (r *ReconcileBareMetalHost) saveHostStatus(host *metal3v1alpha1.BareMetalHost) error {
+func (r *ReconcileBareMetalHost) saveHostStatus(host *metal3v1alpha1.BareMetalHost) (bool, error) {
```

Comment:
With 2 return values, it would be nice to name them so users of the function understand what they're for.